pvlib/pvsystem.py

```diff
@@ -2873,10 +2873,11 @@ def singlediode(photocurrent, saturation_current, resistance_series,
 
         # calculate the IV curve if requested using bishop88
         if ivcurve_pnts:
-            vd = v_oc * (
+            ivcurve_v = v_oc * (
                 (11.0 - np.logspace(np.log10(11.0), 0.0, ivcurve_pnts)) / 10.0
             )
-            ivcurve_i, ivcurve_v, _ = _singlediode.bishop88(vd, *args)
+            ivcurve_i = _singlediode.bishop88_i_from_v(ivcurve_v, *args,
```

Comment:
These changes should be reverted. `singlediode.bishop88` returns the current and voltage vectors. The returned voltage vector is not equal to the input `vd` which is the diode voltage.

Arguably, the returned voltage could be constructed in the way proposed here (compute voltage, not diode voltage, and find the corresponding currents). But that should be discussed in a separate issue.

Reply:
I reverted these changes and created issue #1578 for this. The test cases for the `brentq` and `newton` methods check the accuracy of the currents returned at the voltages of `vd`, so they will not pass. Since these unit tests depend on what voltage values are returned by `pvsystem.singlediode`, I think they should be removed for now and added when issue #1578 is resolved.